packages/ui-kit/src/components/Select/Select.tsx

```diff
@@ -28,20 +31,27 @@ export interface SelectProps {
     | "optionIcon"
     | "option"
   >;
-  onChange?: (option: Option) => void;
+  onChange: SelectOnChangeHandler<TData>;
 }
 
-export const Select: FC<SelectProps> = ({
-  selected,
+export const Select = <TData extends string = string>({
+  selectedValue,
   options,
   error,
   disabled,
   classNames,
   onChange,
-}) => {
+}: SelectProps<TData>) => {
+  const selectedOption: Option<TData> | undefined = options.find(
+    ({ value }) => value === selectedValue
+  );
```

Comment:
It looks like options is already typed so the type should be inferred:
```suggestion
  const selectedOption = options.find(
    ({ value }) => value === selectedValue
  );
```